protocols/request-response/src/lib.rs

```diff
@@ -696,6 +723,17 @@ where
                 }
             }
         }
+        // Its possible that this dial failure is for an existing connection.
+        // If so we need to remove the connection.
+        self.remove_connection(connection_id, peer_id);
+    }
+    fn on_listen_failure(&mut self, ListenFailure { connection_id, .. }: ListenFailure) {
+        // Its possible that this listen failure is for an existing connection.
+        // If so we need to remove the connection.
+        //
+        // TODO: Once  https://github.com/libp2p/rust-libp2p/pull/4818 merges we should pass in the
```

Comment:
Stale comment. #4818 is unrelated now that we agreed to just always work based off `ConnectionId`.